sema4ai/vscode-client/src/extension.ts

```diff
@@ -1088,3 +1102,22 @@ export async function openDataSourceDefinition(entry?: RobotEntry) {
         );
     }
 }
+
+async function checkRuffErrors(folderPath: string): Promise<boolean> {
+    const ruffError: any = await langServer.sendRequest("checkRuffErrors", { folder_path: folderPath });
+    if (!ruffError.success) {
+        vscode.window.showErrorMessage(`Error checking for linting errors: ${ruffError.message}`);
```

Comment:
You should do this a modal dialog and ask if the user if he wants to proceed or show error...

The options are something as:
`Ignore and Publish`, `Cancel and Show Error` (if you can show the error, otherwise just `Cancel`).

If he proceeds you can just return `true` and let him publish, otherwise if you have the errors show them.